### PR TITLE
Update button labels in Data collection wizard

### PR DESCRIPTION
**Pull request by phillipbarron**
## What does this change?

Clicking `Next`, actually saves the state. This might not be clear as clicking in the stepNav does not. This updates the button labels be better signal to the user what is actually happening. 

Clicking `Back` also saved changes - this is probably not what isers would expect. It also causes validation to run, making users feel stuck on that step. This change also removes that call to save and associated validation

## How to test

Not much to test, just a  label change

## Have we considered potential risks?

Always

## Images

Screenshot 2023-07-27 at 08 52 45

### CODE REVIEW

**Comment by dblatcher** [comment]
This is a good idea - the only thing is that clicking the 'back' buttons also saves the state, which might not be obvious when next to the explicit "save and continue" buttons. 

We could change the behaviour of the back buttons to no save the state by changing their `executeStep` property from `executeModify` to `executeSkip` or change the labels to reflect the current action 🤔

**Comment by phillipbarron** [comment]
> This is a good idea - the only thing is that clicking the 'back' buttons also saves the state, which might not be obvious when next to the explicit "save and continue" buttons.
> 
> We could change the behaviour of the back buttons to no save the state by changing their `executeStep` property from `executeModify` to `executeSkip` or change the labels to reflect the current action 🤔

Yeah you're right - The `executeSkip` _feels_  like the right option, but I'm basing that on gut, not evidence. I feel like clicking `Back` implies  "Don't do anything"

**Comment by dblatcher** [comment]
> > This is a good idea - the only thing is that clicking the 'back' buttons also saves the state, which might not be obvious when next to the explicit "save and continue" buttons.
> > We could change the behaviour of the back buttons to no save the state by changing their `executeStep` property from `executeModify` to `executeSkip` or change the labels to reflect the current action 🤔
> 
> Yeah you're right - The `executeSkip` _feels_ like the right option, but I'm basing that on gut, not evidence. I feel like clicking `Back` implies "Don't do anything"

Sounds like a product/UX question!

**Comment by phillipbarron** [comment]
Validation is run when `executeSkip` is used in the `executeStep` - not immediately clear why 
Screenshot 2023-07-28 at 07 44 57

**Review by alinaboghiu — APPROVED** [review]
Nice!

File: libs/newsletter-workflow/src/lib/steps/newsletterData/categoryLayout.ts [hunk 1]
```diff
@@ -41,13 +41,13 @@ export const categoryLayout: WizardStepLayout<DraftService> = {
 	buttons: {
 		back: {
 			buttonType: 'PREVIOUS',
-			label: 'Back',
+			label: 'Back to previous step',
 			stepToMoveTo: getPreviousOrEditStartStepId,
-			executeStep: executeModify,
+			executeStep: executeSkip,
```
[the remark this answers is not in the document]
Comment (phillipbarron) [reply libs/newsletter-workflow/src/lib/steps/newsletterData/categoryLayout.ts:46]: Yes I will - thanks for the review 👍

**Pull request merged**